### ISSUE TITLE
different pipeline versions fails the test runs at different points

### ISSUE THREAD

**Issue by luciazifcakova** [nf-core/mag#717]
### Description of the bug

I have run these comands with these results:

nextflow run nf-core/mag -r 3.2.0 -profile test --outdir . [nf-core/mag] Pipeline completed with errors-
ERROR ~ Error executing process > 'NFCORE_MAG:MAG:BOWTIE2_PHIX_REMOVAL_BUILD (GCA_002596845.1_ASM259684v1_genomic.fna.gz)'

Caused by:
  Process NFCORE_MAG:MAG:BOWTIE2_PHIX_REMOVAL_BUILD (GCA_002596845.1_ASM259684v1_genomic.fna.gz) terminated with an error exit status (127)

Command executed:

  mkdir bowtie
  bowtie2-build --threads 1 GCA_002596845.1_ASM259684v1_genomic.fna.gz GCA_002596845
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_MAG:MAG:BOWTIE2_PHIX_REMOVAL_BUILD":
      bowtie2: $(echo $(bowtie2 --version 2>&1) | sed 's/^.*bowtie2-align-s version //; s/ .*$//')
  END_VERSIONS

Command exit status:
  127

Command output:
  (empty)

Command error:
  .command.sh: line 9: bowtie2-build: command not found

Work dir:
  /flash/MillerU/water_jan/work/5d/756ca0375bc0fdd1bf5fa407d7719a

Tip: you can replicate the issue by changing to the process work dir and entering the command bash .command.run

 -- Check '.nextflow.log' file for details
ERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting

 -- Check '.nextflow.log' file for details 

nextflow run nf-core/mag -r 3.1.0 -profile test --outdir . [nf-core/mag] Pipeline completed with errors-
ERROR ~ Error executing process > 'NFCORE_MAG:MAG:KRONA_KRONADB'

Caused by:
  Process NFCORE_MAG:MAG:KRONA_KRONADB terminated with an error exit status (127)

Command executed:

  ktUpdateTaxonomy.sh \
       \
      taxonomy/
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_MAG:MAG:KRONA_KRONADB":
      krona: 2.7.1
  END_VERSIONS

Command exit status:
  127

Command output:
  (empty)

Command error:
  .command.sh: line 4: ktUpdateTaxonomy.sh: command not found

Work dir:
  /flash/MillerU/water_jan/work/d7/9aeec28068ef69d947caebc5985264

Tip: you can try to figure out what's wrong by changing to the process work dir and showing the script file named .command.sh

 -- Check '.nextflow.log' file for details
ERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting

 -- Check '.nextflow.log' file for details

### Command used and terminal output

```console
nextflow run nf-core/mag -r 3.2.0 -profile test --outdir .

nextflow run nf-core/mag -r 3.1.0 -profile test --outdir .
```

### System information

HPC
 N E X T F L O W   ~  version 24.10.1
slurm
Singularity
CentOS
nf-core/mag 3.2.0 and 3.1.0

**Comment by jfy133** [comment]
Agree, point 3 in the link above is the critical one here :)

You need to specify where to pull the software from otherwise Nextflow tries to find the tools if they are installed on your system already. In this case it seems not.

**Comment by luciazifcakova** [comment]
I have loaded all necessary modules but the mistake was that I did not include in -profile oist tag. After running it again with oist and test tags, only one task have failed.

The failed task was NFCORE_MAG:MAG:BINNING:MAXBIN2 and it failed because:
  .exitcode
255

.command.err
WARNING: Skipping mount /apps/free81/singularity/4.1.4/var/singularity/mnt/session/etc/resolv.conf [files]: /etc/resolv.conf doesn't exist in container

.command.out
MaxBin 2.2.7
Input contig: input/MEGAHIT-test_minigut_sample2.contigs.fa
Located abundance file [MEGAHIT-test_minigut_sample2-test_minigut.abund]
Located abundance file [MEGAHIT-test_minigut_sample2-test_minigut_sample2.abund]
Thread: 6
out header: MEGAHIT-MaxBin2-test_minigut_sample2
Searching against 107 marker genes to find starting seed contigs for [input/MEGAHIT-test_minigut_sample2.contigs.fa]...
Running FragGeneScan....
Running HMMER hmmsearch....
Try harder to dig out marker genes from contigs.
Marker gene search reveals that the dataset cannot be binned (the medium of marker gene number <= 1). Program stop.


Is this a problem of input test dataset or the singularity container?

Thank you.
